config/intellij-idea-inspections.xml

```diff
@@ -3409,6 +3413,9 @@
     <inspection_tool class="OnDemandImport" enabled="true" level="ERROR" enabled_by_default="true"/>
     <inspection_tool class="OneWayWebMethod" enabled="false" level="WARNING"
                      enabled_by_default="false"/>
+    <!-- whole purpose of optional is that it may not always be present -->
+    <inspection_tool class="OptionalGetWithoutIsPresent" enabled="false" level="ERROR"
+                     enabled_by_default="false" />
```

Comment:
not good.

```
Optional.get() is called without isPresent() check
PackageObjectFactory
'Optional.get()' without 'isPresent()' check
TokenUtilTest
'Optional.get()' without 'isPresent()' check
```

This is design problem and/or misusage/illusion of Optional class, without such check for presence , there will be exception - https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get-- .
Solution is smth like - https://stackoverflow.com/a/38725479

Reply:
> PackageObjectFactory

Code is `.get().getKey()` so we can't just return `null` either. Also, this code is wrapped by `containsValue` of the same list, so I highly doubt an if/else check would pass code coverage.
I think we should suppress this, you agree?

Reply:
@romani I rewrote the code. We basically rewrote the `containsValue` twice by different means, so I removed the main one and now we just rely on the one using optional.